src/scanner.cc

```diff
@@ -108,6 +111,94 @@ struct Scanner {
     }
   }
 
+  bool is_escapable_sequence(TSLexer *lexer) {
+    // Note: remember to also update the escape_sequence rule in the
+    // main grammar whenever changing this method
+    auto letter = lexer->lookahead;
+
+    if (letter == 'n' ||
+        letter == 'r' ||
+        letter == 't' ||
+        letter == 'v' ||
+        letter == 'e' ||
+        letter == 'f' ||
+        letter == '\\' ||
+        letter == '$' ||
+        letter == '"') {
+      return true;
+    }
+
+    // Hex
+    if (letter == 'x') {
+      advance(lexer);
+      return isxdigit(lexer->lookahead);
+    }
+
+    // Unicode
+    if (letter == 'u') {
+      advance(lexer);
+      return lexer->lookahead == '{';
+    }
+
+    // Octal
+    return isdigit(lexer->lookahead) && lexer->lookahead >= 0 && lexer->lookahead <= 7;
```

Comment:
`lexer->lookahead` would be the a character, not a number, right? So `lexer->lookahead >= 0` always succeeds. Did you mean `lexer->lookahead >= '0'`?

Reply:
This can also be replaced with a call to [`iswdigit`](https://www.cplusplus.com/reference/cwctype/iswdigit/) in `<cwctype>`. As a bonus, this function is available if you were to compile this grammar into WASM with `tree-sitter build-wasm`; the list of such functions is available [here](https://github.com/tree-sitter/tree-sitter/issues/949).

Reply:
Ouch, good catch! I did indeed mean '0' and '7'